dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,597 @@
+    so a rough idea of parameters to sample works.
+
+    For instance, let's say about 200 or 300 hyper-parameters need to be tested
+    to effectively search the possible hyper-parameters and estimators need a
+    little less than 100 epochs to converge. This will mean the dataset should
+    have chunks that are about 1/3rd of the entire dataset, and ``max_iter`` is
+    between 200 and 300.
+
+    If the search space is larger or more complex, choose to evaluate more
+    hyper-parameters. Increase the parameters to sample by a factor of 2 by:
+
+    * increasing ``params_to_sample`` by a factor of 2
+    * decreasing ``chunk_size`` by a factor of 2.
+
+    There are some limitations to the `current` implementation of Hyperband:
+
+    1. The full dataset is requested to be in distributed memory
+    2. The testing dataset should fit in the memory of a single worker
+
+    References
+    ----------
+    .. [1] "Hyperband: A novel bandit-based approach to hyperparameter
+           optimization", 2016 by L. Li, K. Jamieson, G. DeSalvo, A.
+           Rostamizadeh, and A. Talwalkar.  https://arxiv.org/abs/1603.06560
+
+    """
+
+    def __init__(
+        self,
+        estimator,
+        param_distribution,
+        max_iter=81,
+        aggressiveness=3,
+        test_size=None,
+        patience=False,
+        tol=1e-3,
+        random_state=None,
+        scoring=None,
+    ):
+        self.estimator = estimator
+        self.param_distribution = param_distribution
+        self.max_iter = max_iter
+        self.aggressiveness = aggressiveness
+
+        super(HyperbandSearchCV, self).__init__(
+            estimator,
+            param_distribution,
+            max_iter=max_iter,
+            patience=patience,
+            tol=tol,
+            test_size=test_size,
+            random_state=random_state,
+            scoring=scoring,
+        )
+
+    def _get_SHAs(self, brackets):
+
+        patience = _get_patience(self.patience, self.max_iter, self.aggressiveness)
+
+        # This is the first time self.random_state is used after
+        # HyperbandSearchCV.fit is called.
+        seed_start = check_random_state(self.random_state).randint(2 ** 31)
+        self._SHA_seed = seed_start
+
+        # These brackets are ordered by adaptivity; bracket=0 is least adaptive
+        SHAs = [
+            (
+                b,
+                SuccessiveHalvingSearchCV(
+                    self.estimator,
+                    self.param_distribution,
+                    n_initial_parameters=n,
+                    start_iter=r,
+                    adaptive_max_iter=b + 1,
+                    aggressiveness=self.aggressiveness,
+                    max_iter=self.max_iter,
+                    patience=patience,
+                    tol=self.tol,
+                    test_size=self.test_size,
+                    random_state=seed_start + b if b != 0 else self.random_state,
```

Comment:
To make sure I understand.

I pass `random_state=10`, the first SH gets `random_state=10`, and the second gets `seed_start + b`? Why not just always pass `seed_start + b`? Do we care that the first gets the number 10 as the random state?

Reply:
> Do we care that the first gets the number 10 as the random state?

Yes. I care that the least adaptive bracket of Hyperband gets the `random_state` passed in by the user. This allows the least adaptive bracket and IncrementalSearchCV to sample the same parameters. I implement a test for this ([test_same_random_state_same_params][1]).

This makes it clear that HyperbandSearchCV is a fancier version of IncrementalSearchCV, and also allows it to implement a fair comparison. 

[1]:https://github.com/stsievert/dask-ml/blob/e4abd49d2a77c303f169729cb2c77738504cbb00/tests/model_selection/test_hyperband.py#L321-L354